docs/source/data/errors.mdx

```diff
@@ -288,212 +320,263 @@ This results in a response like the following:
+
+```ts
+import { ApolloServer } from '@apollo/server';
+import { startStandaloneServer } from '@apollo/server/standalone';
+import { ApolloServerErrorCode } from '@apollo/server/errors';
 
-```js {4-12}
 const server = new ApolloServer({
   typeDefs,
   resolvers,
-  csrfPrevention: true,
-  cache: "bounded",
-  plugins: [
-    ApolloServerPluginLandingPageLocalDefault({ embed: true }),
-  ],
-  formatError: (err) => {
-    // Don't give the specific errors to the client.
-    if (err.message.startsWith('Database Error: ')) {
-      return new Error('Internal server error');
+  // highlight-start
+  formatError: (formattedError, error) => {
+    // Return a different error message
+    if (
+      formattedError.extensions.code ===
+      ApolloServerErrorCode.GRAPHQL_VALIDATION_FAILED
+    ) {
+      return {
+        ...formattedError,
+        message: "Your query doesn't match the schema. Try double-checking it!",
+      };
     }
 
-    // Otherwise return the original error. The error can also
+    // Otherwise return the formatted error. This error can also
     // be manipulated in other ways, as long as it's returned.
-    return err;
+    return formattedError;
   },
+  // highlight-end
 });
 
-server.listen().then(({ url }) => {
-  console.log(`🚀 Server ready at ${url}`);
-});
+const { url } = await startStandaloneServer(server);
+console.log(`🚀 Server listening at: ${url}`);
 ```
 
-The error instance received by `formatError` (a `GraphQLError`) contains an `originalError` property, which represents the original error thrown in the resolver. You can use this property to obtain the `instanceof` the error class, such as `AuthenticationError` or `ValidationError`:
+As another example, here we return a more generic error whenever the original error's message begins with `Database Error: `:
 
-```js
-  formatError(err) {
-    if (err.originalError instanceof AuthenticationError) {
-      return new Error('Different authentication error message!');
+```ts
+formatError: (formattedError, error) => {
+  // highlight-start
+  if (formattedError.message.startsWith('Database Error: ')) {
+    return { message: 'Internal server error' };
+  }
+  // highlight-end
+
+  // Otherwise return the formatted error.
+  return formattedError;
+},
+```
+
+If you want to access the originally thrown error (without the JSON formatting), you can use `formatError`'s second argument.
+
+For example, if you are using a database package in your app and you'd like your server to do something when it throws a specific type of database error:
+
+```ts
+ formatError: (formattedError, error) => {
+    if (error instanceof CustomDBError) {
+      // do something specific
+    }
+  },
+```
+
+However, if a _resolver_ throws the error, a `GraphQLError` is wrapped around the initially thrown error.
+
+The `unwrapResolverError` function can remove the `GraphQLError` wrapping from a resolver error or return the error unaltered if it isn't from a resolver.
```

Comment:
Let's provide some justification for why this isn't just an annoyance — namely, you can look at fields like `path` on the `GraphQLError`.

Reply:
Great idea! I added some background here:

> Note, if a _resolver_ throws the error, a `GraphQLError` is wrapped around the initially thrown error. This `GraphQLError` neatly formats the error and contains useful fields, such as the `path` where the error occurred.
>
> In you want to remove the outer `GraphQLError` to access the originally thrown error you can use `unwrapResolverError` from `@apollo/server/errors`. The `unwrapResolverError` function can remove the `GraphQLError` wrapping from a resolver error or return the error unaltered if it isn't from a resolver.
>
> So, we can rewrite the above code snippet to work for errors thrown in and outside of resolvers, like so: